### DISCUSSION TITLE
Do I need to uninstall all apps before installing them again via Obtainium?

### DISCUSSION THREAD

**Discussion by xcpn** [ImranR98/Obtainium#1604] [Q&A]
Right now, I only use Obtainium as Track-Only for apps that I cannot just update via Droid-ify (F-Droid repo).

I'm planning on migrating these apps from Droid-ify to Obtainium and keep using Droid-ify only for apps that don't offer standalone APKs.

My problem is that I have apps that I customized and configured a lot, and some of them don't have an easy way to just export settings/data, reinstall and import again.

Can I add the apps in Obtainium and install them on top of the existing ones (kinda like updating them) so I don't lose data, or it's not possible because of different signatures, etc?

**Comment by xcpn** [comment]
I did a test by downloading Tuta Mail via Droid-ify and then adding it in Obtainium.

Obtainium prompted me to update Tuta Mail immediately after I added it and upon tapping 'update' I got a conflict error while it was updating.

So I assume I need to uninstall all the apps and then reinstall them via Obtainium.

**Comment by Mikaela** [comment]
It depends on whether the app you install via Obtainium and Droidify have the same signer or different one. In the case of F-Droid https://f-droid.org/docs/Reproducible_Builds/ may be of interest.

**Comment by xcpn** [comment]
Yeah they don't have the same signatures unfortunately because I downloaded them via the F-Droid repository and F-Droid signs thems with their own keys.

The APKs themselves in GitHub have the Developer's signatures.
